spring-boot-kotlin-io/src/main/kotlin/com/baeldung/nonblockingcoroutines/config/RouterConfig.kt

```diff
@@ -0,0 +1,19 @@
+package com.baeldung.nonblockingcoroutines.config
+
+import com.baeldung.nonblockingcoroutines.handlers.ProductsHandler
+import kotlinx.coroutines.FlowPreview
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.web.reactive.function.server.coRouter
+
+@Configuration
+class RouterConfiguration {
+
+    @FlowPreview
+    @Bean
+    fun productRoutes(productsHandler: ProductsHandler) = coRouter {
+        GET("/", productsHandler::findAll)
+        GET("/{id}", productsHandler::findOne)
```

Comment:
Just so I understand, the router cannot operate on `Flux`, `Flow` or `Mono` - it needs the downstream handler to `suspend` in order to await the actual value?

Reply:
We can also use Flux/Mono or Flow without any problems. Suspending functions are handled by the framework, Here is an example of using Flux/Mono with the DSL router definition. https://github.com/eugenp/tutorials/tree/master/spring-reactive-kotlin